src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportFromFileService.java

```diff
@@ -0,0 +1,197 @@
+        var exerciseJsonPath = retrieveExerciseJsonPath(extractedZipPath);
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        objectMapper.findAndRegisterModules();
+
+        try {
+            return objectMapper.readValue(exerciseJsonPath.toFile(), ProgrammingExercise.class);
+        }
+        catch (IOException e) {
+            throw new BadRequestAlertException("The JSON file for the programming exercise is not valid or was not found.", "programmingExercise", "exerciseJsonNotValidOrFound");
+        }
+    }
+
+    private void checkRepositoriesExist(Path path) throws IOException {
+        List<String> result;
+        try (Stream<Path> stream = Files.walk(path)) {
+            result = stream.filter(Files::isDirectory).map(f -> f.getFileName().toString())
+                    .filter(name -> name.endsWith("-exercise") || name.endsWith("-tests") || name.endsWith("-solution")).toList();
+        }
+
+        if (result.size() != 3) {
+            throw new BadRequestAlertException("The zip file doesn't contain the template, solution or tests repository or they do not follow the naming scheme.",
+                    "programmingExercise", "repositoriesInZipNotValid");
+        }
+
+    }
+
+    private Path retrieveRepositoryDirectoryPath(Path dirPath, String repoType) {
+        List<Path> result;
+        try (Stream<Path> walk = Files.walk(dirPath)) {
+            result = walk.filter(Files::isDirectory).filter(file -> file.getFileName().toString().endsWith("-" + repoType)).toList();
+        }
+        catch (IOException e) {
+            throw new BadRequestAlertException("Could not read the directory", "programmingExercise", "couldnotreaddirectory");
+        }
+        if (result.size() != 1) {
+            throw new IllegalArgumentException(
+                    "There are either no or more than one sub-directories containing " + repoType + " in their name. Please make sure that there is exactly one.");
+        }
+
+        return result.get(0);
+    }
+
+    private Path retrieveExerciseJsonPath(Path dirPath) throws IOException {
+        List<Path> result;
+        try (Stream<Path> stream = Files.walk(dirPath)) {
+            // if we do not convert the file name to a string, the second filter always returns false
+            // for the third filter, we need to convert it to a string as well as a path doesn't contain a file extension
+
+            result = stream.filter(Files::isRegularFile).filter(file -> file.getFileName().toString().startsWith("Exercise-Details"))
+                    .filter(file -> file.toString().endsWith(".json")).toList();
+        }
+
+        if (result.size() != 1) {
+            throw new BadRequestAlertException("There are either no or more than one json file in the directory!", "programmingExercise", "exerciseJsonNotValidOrFound");
```

Comment:
```suggestion
            throw new BadRequestAlertException("There are either no json files or more than one json file in the directory!", "programmingExercise", "exerciseJsonNotValidOrFound");
```

I think the error message wording is slightly unclear